server/commands/write.go

```diff
@@ -87,6 +113,72 @@ func (c *WriteCommand) validateTuplesets(ctx context.Context, req *openfgapb.Wri
 	return nil
 }
 
+// validateTypesForTuple makes sure that when writing a tuple, the types are compatible.
+// 1. If the tuple is of the form (person:bob, reader, doc:budget), then the type "doc", relation "reader" allows type "person".
```

Comment:
(user=person:bob, relation=reader, object=doc:budget)

Reply:
Or let's agree to use the `object#relation@user` notation from the paper everywhere to keep things shorter 😬

Reply:
Haha @craigpastro i don't want to introduce a brand new notation :P